test/parallel/test-no-enter-tickcallback.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+var allsGood = false;
+var cntr = 0;
+
+process.on('exit', () => {
+  assert.ok(cntr > 0, '_tickDomainCallback was never called');
+});
```

Comment:
If you think being stricter is okay, we could just wrap the fake `_tickDomainCallback` in `common.mustCall(·, 2)` (You can probably judge whether the number of calls is likely to change).